src/blog/2024/08/opentelemetry-with-node-red.md

```diff
@@ -0,0 +1,108 @@
+_Screenshot showing the configuration of opentelmetry node_
+
+1. Drag an `OTEL` node onto the canvas.
+2. Double-click on the node and set the URL to your exporter endpoint (e.g., `http://localhost:4318/v1/traces` for a locally running Jaeger exporter). Provide a name for the service according to your preference, and set the Prefix, which will be added to the root Node-RED span name before the initial node name (you can keep it as "Message" if preferred).
+3. In the Ignore field, add the names of nodes you want to exclude from OpenTelemetry tracing.
+4. In the Propagate field, add the names of nodes to which you want Open Telemetry to forward trace headers for tracking.
+5. Set the Timeout to define how long (in seconds) the OTEL node should wait before ending and discarding a message that has not been modified.
+6. Now deploy the flow by clicking on the top-right deploy button.
+
+Once the flow is deployed, Open Telemetry will start collecting and sending trace data to your specified exporter.
+
+## Monitoring Performance Using the Exporter Web UI
+
+Now, let's monitor the performance and latency between each node to identify delays. For this section, I am assuming you have Jaeger running as your exporter.
+
+1. Open the Jaeger web UI in your browser. By default, it will be available at `http://localhost:16686/`.
+2. Navigate to the "Search" by clicking on the "Search" option at the top.
+3. Select the service name that you configured in the OTEL node from the service field. Once selected, you will see all the traces for each interaction in the flow. You can filter the traces by specific nodes using the operation field.
+4. To monitor and find issues, select the desired trace and click on the "Find Trace" button. Click on the first trace to examine it.
+
+Once the trace opens, you will see the duration taken by each node to process and pass data. Notice the delay in the Kelvin to Celsius function, which is taking 2 seconds. By clicking on the green line corresponding to this function, you can view more detailed information about the trace.
+
+!["Image showing the total duration taken by the flow"](./images/before.png "Image showing the total duration taken by the flow")
+_Image showing the tototal duration taken by the flow_
+
+Since the issue was identified in the function node, you can update the function to remove the delay which we have intentionally added. Replace the current function node with the following JavaScript code:
+
+```js
+// Extract the Kelvin value from the incoming message payload
+let kelvin = msg.payload;
+
+// Function to convert Kelvin to Celsius
+function kelvinToCelsius(kelvin) {
+    return kelvin - 273.15;
+}
+
+// Convert Kelvin to Celsius
+let celsius = kelvinToCelsius(kelvin);
+
+// Set the result in the message payload
+msg.payload = celsius;
+
+// Return the modified message object
+return msg;
+```
```

Comment:
What's the low code equivalent? Can't the chnage node do this?

Reply:
Done! I have replaced the function node with a change node and added a delay node for the delay.